dist/App.js

```diff
@@ -80,16 +80,18 @@ var App = function App(props) {
     selectedTab = retrieveSelectedTab + distinguisher;
   }
 
+  var illegalCharacters = /\W+/gi;
+  selectedTab = selectedTab.replace(illegalCharacters, "-");
```

Comment:
https://github.com/datopian/data-explorer/blob/master/src/App.js#L46
@mpolidori  can we do directly on random function. I think this way the code will look much cleaner. 
```
var distinguisher = '-' + Math.random().toString(36).slice(2,5).replace(illegalCharacters, "-")
```

Reply:
Ah, it's not for characters in that string. It's for a few edge cases in `widget.name`

Reply:
@sagargg ^

Reply:
Maybe I'm missing something but I think `widget.name` will only have chars. `Table`, `Map`, `Chart`.